### PR TITLE
Update snap base to core24

### PR DESCRIPTION
**Pull request by berkayoz**
*The behaviour of automated elf patching changed in core22, the default is to [not patch auto](https://snapcraft.io/docs/linters-classic#heading--issues-auto)
*Snapcraft has changed in core24 to use the craft-application framework to align with other *craft tools, so some things [have changed](https://forum.snapcraft.io/t/micro-howto-migrate-to-core24/39393)
*Currently only can build with build-base: devel and grade: devel

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -45,11 +43,13 @@ parts:
       - pkg-config
       - rsync
       - tcl
+      - sudo
```
Comment (neoaggelos) [inline snap/snapcraft.yaml:46]: why do we need sudo ?

**Review by neoaggelos — COMMENTED** [review]
Looks mostly fine. What's the trade-off of not including `[enable-patchelf]` as a build attribute? I would prefer keeping it out of the parts when it's not required.

Also, please make sure to rebase the strict patch on top of the introduced snapcraft.yaml changes

File: snap/snapcraft.yaml (hunk 1 above)
Comment (berkayoz) [reply snap/snapcraft.yaml:46]: The new build bases (that comes with the craft-application update) does not have sudo installed by default and our Makefile(via build scripts) call it

**Comment by eaudetcobello** [comment]
Closing for now, we can reopen when needed.

**Pull request closed without merging**